src/prommis/uky/uky_flowsheet.py

```diff
@@ -2610,12 +2608,11 @@ def add_costing(flowsheet):
 
     # Leaching costs
     # 4.2 is UKy Leaching - Polyethylene Tanks
-    tank_volume = value(
-        units.convert(flowsheet.fs.leach.volume[0, 1], to_units=units.gal)
-    )
     L_pe_tanks_accounts = ["4.2"]
     m.fs.L_pe_tanks = UnitModelBlock()
-    m.fs.L_pe_tanks.capacity = Var(initialize=tank_volume, units=units.gal)
+    m.fs.L_pe_tanks.capacity = Var(
```

Comment:
Why are you creating a new `Var` for this, and not just using the leach tank volume? This means the costing is disconnected from the flowsheet.

Reply:
@andrewlee94 @bpaul4 So to be clear, how should I handle using values from the flowsheet in the costing? For this particular example, it works if I write `m.fs.L_pe_tanks.capacity = flowsheet.fs.leach.volume[0, 1]`, but for other variables, I've encountered a few problems with this approach when it comes to the flow rates:

1.) In some cases, trying to set `m.fs.RSX_pump.feed_rate = units.convert(
            flowsheet.fs.solex_rougher_load.mscontactor.aqueous_inlet.flow_vol[0],
            to_units=units.gal / units.min,
        )`
results in the DOFs not being equal to zero since it forces me to unfix this line `m.fs.RSX_pump.feed_rate.fix()` as a result of the following error: `AttributeError: 'MonomialTermExpression' object has no attribute 'fix'`

2.) In other flow rate cases, I set the variable as done above and unfix the variable but no DOF errors occur (which doesn't really make sense to me)

3.) If I try to set the variables without doing the unit conversion (some variables do not require unit conversions), then I see the following error: 
`RuntimeError: Re-assigning the component 'flow_vol' from block 'fs.solex_rougher_load.mscontactor.aqueous_inlet_state[0.0]' to
block 'fs.RSX_pump' as 'feed_rate'.

This behavior is not supported by Pyomo; components must have a
single owning block (or model), and a component may not appear
multiple times in a block.  If you want to re-name or move this
component, use the block del_component() and add_component() methods.`

Reply:
I am nit sure what exactly is going on here, but what I can say is you are going about it the wrong way.

First `m.fs.L_pe_tanks.capacity = flowsheet.fs.leach.volume[0, 1]` will not work as expected; you cannot just attach a Pyomo component to a new object and expect it to work.

1. That error implies that whatever you are trying to fix is not a `Var` but and `Expression` or `Reference`. This will not work for obvious reasons. Additionally, you do not need an explicit convert when using `fix` or `set_value` - it will handle the conversion for you if you provide a value with units.
3. That is the result of what I said above - you cannot assign a Pyomo component to another Block.

However, this is mostly beside the point. There should be no need to create new Vars and Blocks here - you should just connect the costing to the existing unit model.